src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetPackageSearchService.cs

```diff
@@ -373,7 +374,15 @@ private async ValueTask<IInstalledAndTransitivePackages> GetInstalledAndTransiti
 
             if (itemFilter == ItemFilter.Installed)
             {
-                packageFeeds.mainFeed = new InstalledPackageFeed(installedPackageCollection, metadataProvider, logger);
+                if (await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(cancellationToken))
+                {
+                    packageFeeds.mainFeed = new InstalledAndTransitivePackageFeed(installedPackageCollection, transitivePackageCollection, metadataProvider);
```

Comment:
Can you calculate the transitivePackageCollection *only* when the experment is enabled?
Right now, this is an unnecessary perf regression.
Line 350.

Reply:
RecommenderFeed also needs `transitivePackageCollection`. That's why I left that code as-is.

Reply:
oh my bad then. 

Resolving this comment as it doesn't apply.